doc/api/vm.md

```diff
@@ -167,10 +167,18 @@ const contextifiedSandbox = vm.createContext({ secret: 42 });
     in stack traces produced by this `Module`.
   * `columnOffset` {integer} Specifies the column number offset that is
     displayed in stack traces produced by this `Module`.
-  * `initalizeImportMeta` {Function} Called during evaluation of this `Module`
+  * `initializeImportMeta` {Function} Called during evaluation of this `Module`
     to initialize the `import.meta`. This function has the signature `(meta,
     module)`, where `meta` is the `import.meta` object in the `Module`, and
     `module` is this `vm.SourceTextModule` object.
+  * `importModuleDynamically` {Function} Called during evaluation of this
+    module when `import()` is called. This function has the signature
+    `(specifier, module)` where `specifier` is the specifier passed to
+    `import()` and `module` is this `vm.SourceTextModule`. If this option is
+    not specified, calls to `import()` will reject with
+    [`ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING`][]. This method can return a
+    Module Namespace, but returning a `vm.SourceTextModule` is recommended in
+    order to take advantage of error tracking.
```

Comment:
nit: Module Namespace Object (maybe also adding a link to MDN if they have this, or the JavaScript spec otherwise)

We still need to deal with the issue of `then` export. I see two paths here:

1. Note here, that if the returned Module Namespace Object has a `then` property then `importModuleDynamically` could throw an error, because the unwrapped object could be something other than a Module Namespace Object. This would be another reason why returning a `vm.SourceTextModule` is preferred.

2. Force users to return an object with a `.namespace` property that is a Module Namespace Object, and validate that. (We still need to keep the `vm.SourceTextModule` branch for error handling, however.)